docs/scaling/SYSTEM.md

```diff
@@ -0,0 +1,190 @@
+
+The **activator** is a globally shared deployment, that is very scalable. Its
+main purposes are **buffering requests** and **reporting metrics** to the
+autoscaler.
+
+The activator is mainly involved around scale to/from zero. When a revision is
+scaled to zero instances, the activator will be put into the data path instead
+of revision’s instances. If requests would hit this revision, the activator
+buffers these requests, pokes the autoscaler with metrics and holds the requests
+until instances of the application appear. As soon as that is the case, the
+activator will forward the requests it buffered to the new instances while
+carefully avoiding to overload the existing instances of the application. The
+activator may also be taken off the data-path for minimal network overhead at
+steady state.
+
+Unlike the queue-proxy, the activator actively sends metrics to the autoscaler
+via a websocket connection to minimize scale-from-zero latencies as much as
+possible.
+
+## APIs
+
+### PodAutoscaler (PA)
+
+**API:** `podautoscalers.autoscaling.internal.knative.dev`
+
+An abstraction for all possible PodAutoscalers. The default implementation is
+the **Knative Pod Autoscaler (KPA)**. There is an adapter that implements
+Kubernetes’ HPA via this abstraction as well.
+
+The PodAutoscaler controls the scaling target, the metric that’s scaled on and
+any other input that is relevant for the autoscaling decider or collector.
+
+PodAutoscalers are automatically created from Revisions by default.
+
+### Metric
+
+**API:** `metrics.autoscaling.internal.knative.dev`
+
+Metric is essentially the API surface to control the collector of the
+autoscaler. It controls which service to scrape, how to aggregate the data and
+more.
+
+Metrics are automatically created from PodAutoscalers by default.
+
+### Decider
+
+**API:** Only exists in-memory today. Only here for completeness sake.
+
+### ServerlessServices (SKS)
+
+**API:** `serverlessservices.networking.internal.knative.dev`
+
+An abstraction on top of Kubernetes Services that controls the data flow and
+especially the switch between having the activator as a buffer on the data path
+(for example when scaled to zero) or routing to the application instances
+directly (when scaled above zero).
+
+This is achieved by creating two Kubernetes services for each revision: A
+**public **service and a **private **service.
+
+The private service is a standard Kubernetes service. Its selector is set to
+point to the application instances deployed and as the deployment grows and
+shrinks, the list of available IPs will do so too.
+
+The public service is an “unmanaged” Kubernetes service. It has no selector and
+therefore does not automatically get endpoint management as the private service
+does. The public service’s endpoints are managed by the SKS reconciler directly.
+
+The SKS has two modes: **Proxy** and **Serve**.
+
+In mode “Serve”, the public service’s endpoints are exactly the same as those of
```

Comment:
```suggestion
In  `Serve` mode, the public service endpoints are exactly the same as those of
```
I think "public service endpoints" is also correct here and generates less stutter.